src/thunderbots/software/ai/hl/stp/evaluation/indirect_chip.h

```diff
@@ -0,0 +1,121 @@
+     */
+    std::optional<Point> indirect_chip_and_chase_target(const World& world);
+
+    /**
+     * Returns the target point that the chipper will shoot at and the chaser will meet
+     * ball at. Given the filtered vector of triangles, determines if the triangles are
+     * empty and if the largest triangle is within reach. The target is where ball will
+     * land according to chipping calibration.
+     *
+     * @param triangles A vector of triangles that is already filtered
+     * @param ball_position Position of the ball
+     *
+     * @return Point to chip and chase at; If empty, point is not valid
+     */
+    std::optional<Point> indirect_chip_and_chase_target(
+        const std::vector<Triangle>& triangles, Point ball_position);
+
+    /**
+     * Creates a vector of triangles that picks all permutations of points from the
+     * list of all non-goalie enemy players as well as the four points returned by
+     * get_chip_area_target_corners.
+     *
+     * @param World Object
+     * @param enemy_players
+     *
+     * @return Vector of triangles
+     */
+    std::vector<Triangle> get_all_triangles(const World& world,
+                                            std::vector<Point> enemy_players);
+
+    /**
+     * Given a vector of triangles, returns a new vector that only contains triangles
+     * without enemy robots.
+     *
+     * @param triangles
+     * @param enemy_players
+     *
+     * @return Vector of triangles with no enemy robots
+     */
+    std::vector<Triangle> filter_open_triangles(std::vector<Triangle> triangles,
+                                                std::vector<Point> enemy_players);
+
+    /**
+     * Returns the center point and area of the given triangle.
+     *
+     * @param triangle
+     *
+     * @return Centre point of triangle
+     * @return Area of triangle
+     */
+    std::pair<Point, double> get_triangle_center_and_area(Triangle triangle);
+
+    /**
+     * Remove all Triangles in a given list whose centers do not fall
+     * within the rectangle returned by get_chip_target_area.
+     *
+     * @param World Object
+     * @param triangles
+     *
+     * @return Valid triangles
+     */
+    std::vector<Triangle> remove_outofbounds_triangles(const World& world,
+                                                       std::vector<Triangle> triangles);
+
+    /**
+     * Returns four Points representing a rectangular region to chip and
+     * chase. This rectangle is from ball's position to enemy's end of field.
+     * All points are 'inset' distance away from each edge of the field to allow a
+     * buffer for catching and prevent ball from leaving the field.
+     *
+     * @param World Object
+     * @param inset Distance away from each edge of field
+     *
+     * @return Four points for rectangle
+     */
+    std::vector<Point> get_chip_target_area_corners(const World& world, double inset);
```

Comment:
If I understand this function correctly, perhaps `findBestChipTargetArea` might be a bit more descriptive?

Reply:
Edited function name as per suggested, although I'm curious as to what constitutes this area to be the "best" region to chip at!

Reply:
Looks like this just constructs a Rectangle where `x` is bounded [ball x position, enemy goal x position] and `y` is bounded by [left side of the field, right side of the field]. I think this just means that we want to chip the ball towards the enemy net, but we're not super concerned about where exactly, as long as it's closer. We could definitely make this more sophisticated later on!

Reply:
Ah, I see! I'd be interested to work on this further in the future. Thanks for the clarification!